src/test/java/hudson/plugins/timestamper/action/TimestampsActionOutputTest.java

```diff
@@ -300,29 +303,6 @@ public void testRead_noTimestamps() throws Exception {
     assertThat(readLines(), is(expectedLines));
   }
 
-  @Test
-  public void testRead_timestampsInLogFileOnly() throws Exception {
-    when(timestampsReader.read()).thenReturn(Optional.empty());
-
-    List<Line> lines = new ArrayList<>();
-    int i = 1;
-    for (Timestamp timestamp : TIMESTAMPS) {
-      Line line = mock(Line.class);
-      when(line.getText()).thenReturn("line" + i);
-      when(line.readTimestamp()).thenReturn(Optional.of(timestamp));
-      lines.add(line);
-      i++;
-    }
-
-    OngoingStubbing<Optional<Line>> nextLineStubbing = when(logFileReader.nextLine());
-    for (Line line : lines) {
-      nextLineStubbing = nextLineStubbing.thenReturn(Optional.of(line));
-    }
-    nextLineStubbing.thenReturn(Optional.empty());
-
-    assertThat(readLines(), is(expectedLines));
-  }
```

Comment:
Did not yet look into what it would take to get this unit test passing again after the refactoring that removed the `Line` class. I doubt it was of much value anyway, since it was using mocks rather than a real Jenkins instance.

Reply:
I have had consistently poor experience with mocking frameworks in Jenkins tests.